### PR TITLE
Make goveralls verbose, match goveralls settings in Makefile

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:
Use prow `PULL_REFS` to indicate the branch for goveralls, so that it can
pick up the right branch.

For goveralls env vars:
- replace `CI_PULL_REQUEST` with `CI_PR_NUMBER`,
- remove `CI_BUILD_NUMBER` as it's not used,
- set debug and verbose flags.

Also set goveralls debug and verbose flags.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes lemurheavy/coveralls-public#1480

**Special notes for your reviewer**:
- goveralls: https://github.com/mattn/goveralls/
- https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dhiller** [comment]
Looking good:

output from test job is:
```
Job ##11938.1
https://coveralls.io/jobs/66968407
```

Status link points to that execution 11938.1: https://coveralls.io/builds/33066398

File: Makefile [hunk 1]
```diff
@@ -57,7 +57,7 @@ coverage:
 	hack/dockerized "./hack/coverage.sh ${WHAT}"
 
 goveralls: go-build
-	SYNC_OUT=false hack/dockerized "COVERALLS_TOKEN_FILE=${COVERALLS_TOKEN_FILE} COVERALLS_TOKEN=${COVERALLS_TOKEN} CI_NAME=prow CI_BUILD_NUMBER=${BUILD_ID} CI_BRANCH=${PULL_BASE_REF} CI_PULL_REQUEST=${PULL_NUMBER} ./hack/goveralls.sh"
+	SYNC_OUT=false hack/dockerized "COVERALLS_TOKEN_FILE=${COVERALLS_TOKEN_FILE} COVERALLS_TOKEN=${COVERALLS_TOKEN} CI_NAME=prow CI_BRANCH=${PULL_REFS} CI_PR_NUMBER=${PULL_NUMBER} ./hack/goveralls.sh"
```
Comment (xpivarc) [inline Makefile:60]: We can also remove COVERALLS_TOKEN, right?

Comment (dhiller) [reply Makefile:60]: No, I'd like to keep this to enable simple manual testing.

**Pull request merged**